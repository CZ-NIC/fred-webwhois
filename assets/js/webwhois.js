/* global gettext */

/*
    Webwhois and Public requests
*/

function emailInRegistry(inputCustomEmail, confirmationMethodSelect, confirmationMethodOptions) {
    inputCustomEmail.value = ''
    inputCustomEmail.disabled = true
    inputCustomEmail.classList.remove('required')
    confirmationMethodSelect.selectedIndex = confirmationMethodOptions.indexOf('signed_email')
    confirmationMethodSelect.disabled = true
}

function customEmail(inputCustomEmail, confirmationMethodSelect) {
    inputCustomEmail.disabled = false
    inputCustomEmail.classList.add('required')
    confirmationMethodSelect.disabled = false
}

document.addEventListener('DOMContentLoaded', () => {
    if (document.querySelector('form.webwhois-public-request')) {
        const form = document.querySelector('form.webwhois-public-request')
        const emailChoice = document.querySelectorAll('input[name=send_to_0]')
        const inputCustomEmail = document.querySelector('input[name=send_to_1]')
        const confirmationMethodSelect = document.querySelector('select[name=confirmation_method]')
        const confirmationMethodOptions = []

        Array.from(confirmationMethodSelect.options).forEach(option => {
            confirmationMethodOptions.push(option.value)
        })

        emailChoice.forEach(email => {
            email.addEventListener('click', (e) => {
                e.target.value === 'email_in_registry' ?
                    emailInRegistry(inputCustomEmail, confirmationMethodSelect, confirmationMethodOptions) :
                    customEmail(inputCustomEmail, confirmationMethodSelect)
            })

            if (email.checked && email.value === 'email_in_registry') {
                emailInRegistry(inputCustomEmail, confirmationMethodSelect, confirmationMethodOptions)
            } else if (email.checked) {
                customEmail(inputCustomEmail, confirmationMethodSelect)
            }
        })

        document.querySelector('input[name=handle]').classList.add('required')
        form.addEventListener('submit', (e) => {
            document.querySelectorAll('.errorlist').forEach(error => error.remove())
            document.querySelectorAll('input.required').forEach(input => {
                if (!input.value) {
                    input.insertAdjacentHTML('beforebegin', `
                    <div class='errorlist'>${gettext('This field is required.')}</div>
                `)
                    e.preventDefault()
                }
            })
        })
    }
})
