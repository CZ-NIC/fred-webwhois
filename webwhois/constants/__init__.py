#
# Copyright (C) 2018-2021  CZ.NIC, z. s. p. o.
#
# This file is part of FRED.
#
# FRED is free software: you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation, either version 3 of the License, or
# (at your option) any later version.
#
# FRED is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU General Public License for more details.
#
# You should have received a copy of the GNU General Public License
# along with FRED.  If not, see <https://www.gnu.org/licenses/>.
#
"""Webwhois constants."""
from django.utils.translation import ugettext_lazy as _

from .utils import _DnskeyAlgorithmBase

STATUS_LINKED = 'linked'
STATUS_DELETE_CANDIDATE = 'deleteCandidate'

STATUS_VERIFICATION_IN_PROCESS = 'contactInManualVerification'
STATUS_VERIFICATION_PASSED = 'contactPassedManualVerification'
STATUS_VERIFICATION_FAILED = 'contactFailedManualVerification'

STATUS_CONDITIONALLY_IDENTIFIED = 'conditionallyIdentifiedContact'
STATUS_IDENTIFIED = 'identifiedContact'
STATUS_VALIDATED = 'validatedContact'

SEND_TO_IN_REGISTRY = 'email_in_registry'
SEND_TO_CUSTOM = 'custom_email'


# https://www.iana.org/assignments/dns-sec-alg-numbers/dns-sec-alg-numbers.xhtml
_DNSKEY_ALGORITHMS = {
    'DELETE_DS': (0, _("Delete DS")),
    'RSAMD5': (1, _("RSA/MD5 (deprecated, see 5)")),
    'DH': (2, _("Diffie-Helman")),
    'DSA': (3, _("DSA/SHA1")),
    'RSASHA1': (5, _("RSA/SHA-1")),
    'DSA_NSEC3_SHA1': (6, _("DSA-NSEC3-SHA1")),
    'RSASHA1_NSEC3_SHA1': (7, _("RSASHA1-NSEC3-SHA1")),
    'RSASHA256': (8, _("RSA/SHA-256")),
    'RSASHA512': (10, _("RSA/SHA-512")),
    'GOST': (12, _("GOST R 34.10-2001")),
    'ECDSAP256SHA256': (13, _("ECDSA Curve P-256 with SHA-256")),
    'ECDSAP384SHA384': (14, _("ECDSA Curve P-384 with SHA-384")),
    'ED25519': (15, _("Ed25519")),
    'ED448': (16, _("Ed448")),
    'INDIRECT': (252, _("Reserved for Indirect Keys")),
    'PRIVATEDNS': (253, _("Private algorithm")),
    'PRIVATEOID': (254, _("Private algorithm OID")),
}
# Add usassigned
_DNSKEY_ALGORITHMS.update({'UNASSIGNED_{}'.format(i): (i, _("Unassigned")) for i in range(17, 123)})
# Add reserved
_DNSKEY_ALGORITHMS.update({'RESERVED_{}'.format(i): (i, _("Reserved")) for i in range(0, 256)
                           if i not in [v[0] for v in _DNSKEY_ALGORITHMS.values()]})
DnskeyAlgorithm = _DnskeyAlgorithmBase('DnskeyAlgorithm', _DNSKEY_ALGORITHMS)  # type: ignore[arg-type]
