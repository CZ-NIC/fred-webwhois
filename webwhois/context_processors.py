#!/usr/bin/python
#
# Copyright (C) 2021  CZ.NIC, z. s. p. o.
#
# This file is part of FRED.
#
# FRED is free software: you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation, either version 3 of the License, or
# (at your option) any later version.
#
# FRED is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU General Public License for more details.
#
# You should have received a copy of the GNU General Public License
# along with FRED.  If not, see <https://www.gnu.org/licenses/>.
#
from functools import lru_cache
from typing import Any, Dict, List, Tuple, cast

from django.http import HttpRequest

from .utils import WHOIS


@lru_cache()
def _get_managed_zones() -> Tuple[str, ...]:
    """Return managed zones."""
    return tuple(cast(List[str], WHOIS.get_managed_zone_list()))


def managed_zones(request: HttpRequest) -> Dict[str, Any]:
    """Return a context with managed zones."""
    return {
        'managed_zones': _get_managed_zones(),
    }
