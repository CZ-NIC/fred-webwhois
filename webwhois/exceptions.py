#
# Copyright (C) 2021-2022  CZ.NIC, z. s. p. o.
#
# This file is part of FRED.
#
# FRED is free software: you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation, either version 3 of the License, or
# (at your option) any later version.
#
# FRED is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU General Public License for more details.
#
# You should have received a copy of the GNU General Public License
# along with FRED.  If not, see <https://www.gnu.org/licenses/>.
#
import warnings
from typing import Any, Optional

_DEPRECATED_MSG = "Dictionary interface on 'server_exception' variable is deprecated. Use attributes instead."


class WebwhoisError(Exception):
    """Represents an error in searching an object from Whois.

    Attributes:
        code: An error code.
        title: A human readable error title.
        message: A human readable error message.
    """

    def __init__(self, code: str, *, title: str, message: Optional[str] = None, **kwargs):
        self.code = code
        self.title = title
        self.message = message
        # Data for backwards compatible dictionary interface.
        self._data = dict(code=code, title=title, message=message, **kwargs)
        super().__init__(code, title, message)

    def __getitem__(self, key: Any) -> Any:
        warnings.warn(_DEPRECATED_MSG, DeprecationWarning)
        return self._data[key]

    def get(self, key: Any, default: Any = None) -> Any:
        """Return a value or default."""
        warnings.warn(_DEPRECATED_MSG, DeprecationWarning)
        return self._data.get(key, default)
