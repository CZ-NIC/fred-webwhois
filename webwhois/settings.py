#
# Copyright (C) 2016-2022  CZ.NIC, z. s. p. o.
#
# This file is part of FRED.
#
# FRED is free software: you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation, either version 3 of the License, or
# (at your option) any later version.
#
# FRED is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU General Public License for more details.
#
# You should have received a copy of the GNU General Public License
# along with FRED.  If not, see <https://www.gnu.org/licenses/>.
#
import os
from functools import partial
from typing import Any, Dict

from appsettings import AppSettings, DictSetting, FileSetting, StringSetting
from frgal import make_credentials


class LoggerOptionsSetting(DictSetting):
    """Custom dict setting for logger options."""

    def transform(self, value: Dict[str, Any]) -> Dict[str, Any]:
        """Transform the credentials."""
        value = super().transform(value)
        if 'credentials' in value:
            value['credentials'] = make_credentials(**value['credentials'])
        return value


class WebwhoisAppSettings(AppSettings):
    """Web whois settings."""

    CDNSKEY_NETLOC = StringSetting(default=None)
    CDNSKEY_SSL_CERT = FileSetting(default=None, mode=os.R_OK)
    CORBA_NETLOC = StringSetting(default=partial(os.environ.get, 'FRED_WEBWHOIS_NETLOC', 'localhost'))
    CORBA_CONTEXT = StringSetting(default='fred')
    LOGGER = StringSetting(default='grill.DummyLoggerClient')
    LOGGER_OPTIONS = LoggerOptionsSetting(default={})

    class Meta:
        setting_prefix = 'WEBWHOIS_'


WEBWHOIS_SETTINGS = WebwhoisAppSettings()
