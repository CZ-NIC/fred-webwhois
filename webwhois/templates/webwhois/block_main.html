{% extends base_template %}
{% load i18n %}

{% block content %}
    <div id="whois">
        {% block webwhois_main %}

            {% block webwhois_header %}
                <h1>{% trans "Search in the Registry (Whois)" %}</h1>
                <p>
                    {% trans "Search results for handle" %} <strong class="text-primary">{{ handle }}</strong>:
                </p>
            {% endblock webwhois_header %}

            {% block webwhois_content %}{% endblock webwhois_content %}

            {% block webwhois_footer %}
                <a class="btn btn-primary mt-5" href="{% url "webwhois:form_whois" %}">
                    {% trans "New search in the registry" %}
                </a>
            {% endblock webwhois_footer %}

        {% endblock webwhois_main %}
    </div>
{% endblock content %}
