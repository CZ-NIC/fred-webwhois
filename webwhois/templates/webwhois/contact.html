{% extends "webwhois/block_main.html" %}
{% load i18n static webwhois_filters %}

{% block title %}{% trans "Contact" %} {{ handle }} - {{ block.super }}{% endblock %}

{% block webwhois_header %}
    {{ block.super }}
    <h2>{% trans "Contact details" %}</h2>
{% endblock webwhois_header %}

{% block webwhois_content %}
    {% with contact=registry_objects.contact.detail %}
        <table class="result contact">
            <tr>
                <th>{% trans "Handle" %}</th>
                <td><span class="handle">{{ contact.handle }}</span>{% block webwhois_contact_handle %}{% endblock webwhois_contact_handle %}</td>
            </tr>
            {% if registry_objects.contact.is_linked %}
                <tr>
                    <th>{% trans "Organization" %}</th>
                    <td class="organization">{% if contact.organization.disclose %}
                            {{ contact.organization.value|default_if_none:'' }}
                        {% else %}
                            <span class="not-disclosed">{% trans "Not disclosed" %}</span>
                        {% endif %}
                    </td>
                </tr>
                <tr>
                    <th>{% trans "Name" %}</th>
                    <td class="full-name">{% if contact.name.disclose %}
                            {{ contact.name.value|default_if_none:'' }}
                        {% else %}
                            <span class="not-disclosed">{% trans "Not disclosed" %}</span>
                        {% endif %}
                    </td>
                </tr>
                <tr>
                    <th>{% trans "VAT number" %}</th>
                    <td class="vat-number">{% if contact.vat_number.disclose %}
                            {{ contact.vat_number.value|default_if_none:'' }}
                        {% else %}
                            <span class="not-disclosed">{% trans "Not disclosed" %}</span>
                        {% endif %}
                    </td>
                </tr>
                <tr>
                    <th>{% trans "Identification type" %}</th>
                    <td class="ident-type">{% if contact.identification.disclose %}
                            {{ contact.identification.value.identification_type|contact_ssn_type_label|default_if_none:'' }}
                        {% else %}
                            <span class="not-disclosed">{% trans "Not disclosed" %}</span>
                        {% endif %}
                    </td>
                </tr>
                <tr>
                    <th>{% trans "Identification data" %}</th>
                    <td class="ident-value">
                        {% if contact.identification.disclose %}
                            {% if contact.identification.value.identification_type == "BIRTHDAY" %}
                                {{ registry_objects.contact.birthday|default_if_none:'' }}
                            {% else %}
                                {{ contact.identification.value.identification_data|default_if_none:'' }}
                            {% endif %}
                        {% else %}
                            <span class="not-disclosed">{% trans "Not disclosed" %}</span>
                        {% endif %}
                    </td>
                </tr>
                <tr>
                    <th>{% trans "Email" %}</th>
                    <td class="email">{% if contact.email.disclose %}
                            {% if contact.email.value %}
                                <a href="mailto:{{ contact.email.value }}">{{ contact.email.value }}</a>
                            {% endif %}
                        {% else %}
                            <span class="not-disclosed">{% trans "Not disclosed" %}</span>
                        {% endif %}
                    </td>
                </tr>
                <tr>
                    <th>{% trans "Notification email" %}</th>
                    <td class="notify-email">{% if contact.notify_email.disclose %}
                            {% if contact.notify_email.value %}
                                <a href="mailto:{{ contact.notify_email.value }}">{{ contact.notify_email.value }}</a>
                            {% endif %}
                        {% else %}
                            <span class="not-disclosed">{% trans "Not disclosed" %}</span>
                        {% endif %}
                    </td>
                </tr>
                <tr>
                    <th>{% trans "Phone" %}</th>
                    <td class="phone">{% if contact.phone.disclose %}
                            {{ contact.phone.value|default_if_none:'' }}
                        {% else %}
                            <span class="not-disclosed">{% trans "Not disclosed" %}</span>
                        {% endif %}
                    </td>
                </tr>
                <tr>
                    <th>{% trans "Fax" %}</th>
                    <td class="fax">
                        {% if contact.fax.disclose %}
                            {{ contact.fax.value|default_if_none:'' }}
                        {% else %}
                            <span class="not-disclosed">{% trans "Not disclosed" %}</span>
                        {% endif %}
                    </td>
                </tr>
                <tr>
                    <th>{% trans "Registered since" %}</th>
                    <td class="registered-since">
                        {{ contact.created|date:"SHORT_DATE_FORMAT" }}
                    </td>
                </tr>
                <tr>
                    <th>{% trans "Created by registrar" %}</th>
                    <td class="creating-registrar">
                        {% if contact.creating_registrar_handle %}
                            <a href="{% url "webwhois:detail_registrar" handle=contact.creating_registrar_handle %}">{{ contact.creating_registrar_handle }}</a> {{ registry_objects.contact.creating_registrar.name }}
                        {% endif %}
                    </td>
                </tr>
                <tr>
                    <th>{% trans "Last update" %}</th>
                    <td class="last-update">{{ contact.changed|date:"SHORT_DATE_FORMAT"|default_if_none:'' }}</td>
                </tr>
                <tr>
                    <th>{% trans "Last transfer" %}</th>
                    <td class="last-transfer">{{ contact.last_transfer|date:"SHORT_DATE_FORMAT"|default_if_none:'' }}</td>
                </tr>
                <tr>
                    <th>{% trans "Address" %}</th>
                    <td class="address">{% if contact.address.disclose %}
                            {% spaceless %}
                                {% if contact.address.value %}
                                    {% with addr=contact.address.value %}
                                        {{ addr.street1 }}{% if addr.street2 %}, {{ addr.street2 }}{% endif %}{% if addr.street3 %}, {{ addr.street3 }}{% endif %}, {{ addr.postalcode }} {{ addr.city }}{% if addr.stateorprovince %} â€“ {{ addr.stateorprovince }}{% endif %}, {{ addr.country_code }}
                                    {% endwith %}
                                {% endif %}
                            {% endspaceless %}
                        {% else %}
                            <span class="not-disclosed">{% trans "Not disclosed" %}</span>
                        {% endif %}
                    </td>
                </tr>
            {% endif %}
            <tr>
                <th>{% trans "Sponsoring registrar" %}</th>
                <td class="sponsoring-registrar">
                    {% if contact.sponsoring_registrar_handle %}
                        <a href="{% url "webwhois:detail_registrar" handle=contact.sponsoring_registrar_handle %}">{{ contact.sponsoring_registrar_handle }}</a> {{ registry_objects.contact.sponsoring_registrar.name }}
                    {% endif %}
                </td>
            </tr>
            {% if registry_objects.contact.is_linked and registry_objects.contact.verification_status %}
                <tr>
                    <th>{% trans "Contact verification status" %}</th>
                    <td class="verification-status">{% for status in registry_objects.contact.verification_status %}
                         <div>
                            <img src="{% static status.icon %}" alt="{{ status.code }}" /> <span>{{ status.label }}</span>
                         </div>
                        {% endfor %}
                    </td>
                </tr>
            {% endif %}
            <tr>
                <th>{% trans "Status" %}</th>
                <td class="status" data-codes="{{ contact.statuses|join:"," }}">
                    {% if registry_objects.contact.is_linked %}
                        {% for description in registry_objects.contact.status_descriptions %}
                            <div>{{ description }}</div>
                        {% endfor %}
                    {% else %}
                        <div>{% trans "Contact is registered but not linked to other registry object" %}</div>
                    {% endif %}
                </td>
            </tr>
        </table>
        {% url "webwhois:record_statement_pdf" object_type="contact" handle=contact.handle as record_statement_pdf_url %}
        {% include "webwhois/verified_record_statement_link.html" %}
    {% endwith %}
{% endblock webwhois_content %}
