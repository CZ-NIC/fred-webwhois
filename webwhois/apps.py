#
# Copyright (C) 2017-2022  CZ.NIC, z. s. p. o.
#
# This file is part of FRED.
#
# FRED is free software: you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation, either version 3 of the License, or
# (at your option) any later version.
#
# FRED is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU General Public License for more details.
#
# You should have received a copy of the GNU General Public License
# along with FRED.  If not, see <https://www.gnu.org/licenses/>.
#
"""AppConfig definition."""
from django.apps import AppConfig

from .settings import WebwhoisAppSettings


class WebwhoisAppConfig(AppConfig):
    name = 'webwhois'

    def ready(self) -> None:
        from webwhois.utils.corba_wrapper import LOGGER

        from .constants import (LOGGER_SERVICE, PUBLIC_REQUESTS_LOGGER_SERVICE, LogEntryType, LogResult,
                                PublicRequestsLogEntryType, PublicRequestsLogResult)

        WebwhoisAppSettings.check()

        LOGGER.client.register_service(LOGGER_SERVICE, handle='webwhois_')
        LOGGER.client.register_log_entry_types(LOGGER_SERVICE, LogEntryType)
        LOGGER.client.register_results(LOGGER_SERVICE, LogResult)

        LOGGER.client.register_service(PUBLIC_REQUESTS_LOGGER_SERVICE, handle='pubreq_')
        LOGGER.client.register_log_entry_types(PUBLIC_REQUESTS_LOGGER_SERVICE, PublicRequestsLogEntryType)
        LOGGER.client.register_results(PUBLIC_REQUESTS_LOGGER_SERVICE, PublicRequestsLogResult)
