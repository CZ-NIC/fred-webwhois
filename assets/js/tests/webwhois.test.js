/* eslint-env jest */
import '../webwhois'

// Globally mock gettext
global.gettext = jest.fn((text) => text);

const LOAD_NODE = `
    <form class="webwhois-public-request">
        <input name="handle" required="">
        <input name="send_to_0" value="email_in_registry" checked required>
        <input name="send_to_0" value="custom_email" required>
        <input  name="send_to_1">
        <select name="confirmation_method">
            <option value="signed_email">Email signed by a qualified certificate</option>
        </select>
    </form>
`

const LOAD_NODE_REGISTRY = `
    <form class="webwhois-public-request">
        <input name="handle" required="" class="required">
        <input name="send_to_0" value="email_in_registry" checked required>
        <input name="send_to_0" value="custom_email" checked>
        <input name="send_to_1" disabled>
        <select name="confirmation_method">
            <option value="signed_email">Email signed by a qualified certificate</option>
        </select>
    </form>
`
const LOAD_NODE_SUBMIT = `
    <form class="webwhois-public-request">
        <div class="errorlist">This field is required.</div>
        <input name="handle" class="required" value="test">
        <input name="handle" class="required" value="">
        <select name="confirmation_method"></select>
    </form>
`

describe('webwhois', () => {
    test('Load page', () => {
        document.dispatchEvent(new Event('DOMContentLoaded'))
        expect(document.body).toMatchSnapshot()
    })

    test('Event listeners click to email_in_registry', () => {
        document.body.innerHTML = LOAD_NODE
        document.dispatchEvent(new Event('DOMContentLoaded'))
        document.querySelector('input[value=email_in_registry]').dispatchEvent(new Event('click'))
    })

    test('Event listeners click to custom_email', () => {
        document.body.innerHTML = LOAD_NODE_REGISTRY
        document.dispatchEvent(new Event('DOMContentLoaded'))
        document.querySelector('input[value=custom_email]').dispatchEvent(new Event('click'))
    })

    test('Event listeners submit', () => {
        document.body.innerHTML = LOAD_NODE_SUBMIT
        document.dispatchEvent(new Event('DOMContentLoaded'))
        document.querySelector('.webwhois-public-request').dispatchEvent(new Event('submit'))
    })
})
