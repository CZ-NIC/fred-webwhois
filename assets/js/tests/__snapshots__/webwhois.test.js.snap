// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`webwhois Load page 1`] = `<body />`;
